Fix -Wincompatible-pointer-types error with gcc 14.

```
src/common.c: In function 'read_wav_sample':
src/common.c:439:46: error: assignment to 'INT16 *' {aka 'short int *'} from
incompatible pointer type 'signed char *' [-Wincompatible-pointer-types]
  439 |                         flac_file.write_data = result->data;
```

--- libretro-mame2003-1a268a7-origin/src/common.c
+++ libretro-mame2003-1a268a7/src/common.c
@@ -69,7 +69,7 @@
 typedef struct _flac_reader
 {
 	UINT8* rawdata;
-	INT16* write_data;
+	signed char* write_data;
 	int position;
 	int length;
 	int decoded_size;
